soccer/MainWindow.cpp

```diff
@@ -974,30 +982,44 @@ void MainWindow::channel(int n) {
 // Simulator commands
 
 void MainWindow::on_actionCenterBall_triggered() {
-    SimCommand cmd;
-    cmd.mutable_ball_pos()->set_x(0);
-    cmd.mutable_ball_pos()->set_y(0);
-    cmd.mutable_ball_vel()->set_x(0);
-    cmd.mutable_ball_vel()->set_y(0);
-    _ui.fieldView->sendSimCommand(cmd);
+    grSim_Packet simPacket;
+    grSim_BallReplacement* ball_replace =
+        simPacket.mutable_replacement()->mutable_ball();
+
+    ball_replace->set_x(0);
+    ball_replace->set_y(0);
+    ball_replace->set_vx(0);
+    ball_replace->set_vy(0);
+
+    _ui.fieldView->sendSimCommand(simPacket);
 }
 
 void MainWindow::on_actionStopBall_triggered() {
-    SimCommand cmd;
-    cmd.mutable_ball_vel()->set_x(0);
-    cmd.mutable_ball_vel()->set_y(0);
-    _ui.fieldView->sendSimCommand(cmd);
+    grSim_Packet simPacket;
+    grSim_BallReplacement* ball_replace =
+        simPacket.mutable_replacement()->mutable_ball();
+
+    Geometry2d::Point ballPos =
+        _ui.fieldView->getTeamToWorld() * state()->ball.pos;
+    ball_replace->set_x(ballPos.x());
+    ball_replace->set_y(ballPos.y());
+    ball_replace->set_vx(0);
+    ball_replace->set_vy(0);
+    _ui.fieldView->sendSimCommand(simPacket);
 }
 
 void MainWindow::on_actionResetField_triggered() {
```

Comment:
Maybe we can keep the reset button working (the reset button is very useful, so I would like to keep it if possible). In the right click menu, there is 'yellow robots->put all inside with formation x' which we could call instead of a real reset function when we click this.